crates/router/src/core/user.rs

```diff
@@ -1142,6 +1167,7 @@ pub async fn create_internal_user(
             UserStatus::Active,
         )
         .add_entity(domain::MerchantLevel {
+            tenant_id: common_utils::consts::DEFAULT_TENANT.to_string(),
```

Comment:
this is for internal user.

Reply:
But if I hit this API in a particular tenant deployment, it will create a internal user in public tenant, which doesn't make sense.

And we may not have control over the admins of other tenants. If you don't want application to create any internal user in other tenants than public, then have a check instead but don't create a user in public tenant if the admin didn't say otherwise.